File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> lookup(nums.begin(), nums.end());
        int max_ans = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (lookup.find(nums[i] - 1) == lookup.end()) {
                int min_ans = 0, j = nums[i];
                while (lookup.find(j) != lookup.end()) {
                    j++;
                    min_ans++;
                }
                max_ans = max(max_ans, min_ans);
            }
        }
        return max_ans;
    }
};

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        while l <= r:
            mid = int((l + r) // 2)
            print(f"{l} {r} {mid}: {nums[mid]}")
            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                l = mid + 1
            else:
                r = mid - 1
        return -1
